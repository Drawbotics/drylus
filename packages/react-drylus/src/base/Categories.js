const Categories = {
  BRAND: 'BRAND',
  SUCCESS: 'SUCCESS',
  INFO: 'INFO',
  WARNING: 'WARNING',
  DANGER: 'DANGER',
};


export default Categories;
