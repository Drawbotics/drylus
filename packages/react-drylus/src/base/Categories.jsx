const Categories = {
  PRIMARY: 'PRIMARY',
  SUCCESS: 'SUCCESS',
  INFO: 'INFO',
  WARNING: 'WARNING',
  DANGER: 'DANGER',
};


export default Categories;
