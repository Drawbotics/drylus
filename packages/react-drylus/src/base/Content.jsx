import React from 'react';
import { css, cx } from 'emotion';
import PropTypes from 'prop-types';
import sv from '@drawbotics/style-vars';


const styles = {
  root: css`
    background: ${sv.backgroundColor};
  `,
  fullHeight: css`
    flex: 1;
  `,
  children: css`
    max-width: ${sv.maxWidthLarge};
    margin: auto;
  `,
  fullWidth: css`
    max-width: none;
  `,
};


const Content = ({ children, fullHeight, fullWidth }) => {
  return (
    <div className={cx(styles.root, {
      [styles.fullHeight]: fullHeight,
    })}>
      <div className={cx(styles.children, { [styles.fullWidth]: fullWidth })}>
        {children}
      </div>
    </div>
  );
};


Content.propTypes = {
  children: PropTypes.node.isRequired,

  /** If true, the content will take all the height available */
  fullHeight: PropTypes.bool,

  /** If true, the content will not be limited to 1200px */
  fullWidth: PropTypes.bool,
};


export default Content;
