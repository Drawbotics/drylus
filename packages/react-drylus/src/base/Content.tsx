import sv from '@drawbotics/drylus-style-vars';
import { css, cx } from 'emotion';
import React from 'react';

import { Style } from '../types';

const styles = {
  root: css`
    background: ${sv.backgroundColor};
  `,
  fullHeight: css`
    flex: 1;
  `,
  children: css`
    max-width: ${sv.maxWidthLarge};
    margin: auto;
  `,
  fullWidth: css`
    max-width: none;
    height: 100%;
  `,
};

interface ContentProps {
  children: React.ReactNode;

  /** If true, the content will take all the height available */
  fullHeight?: boolean;

  /** If true, the content will not be limited to 1200px */
  fullWidth?: boolean;

  /** Used for style overrides */
  style?: Style;
}

export const Content = ({ children, fullHeight, fullWidth, style }: ContentProps) => {
  return (
    <div
      style={style}
      className={cx(styles.root, {
        [styles.fullHeight]: fullHeight,
      })}>
      <div className={cx(styles.children, { [styles.fullWidth]: fullWidth })}>{children}</div>
    </div>
  );
};
