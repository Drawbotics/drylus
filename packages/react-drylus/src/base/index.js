export ThemeProvider from './ThemeProvider';
export Page from './Page';
export Sizes from './Sizes';
export Categories from './Categories';
