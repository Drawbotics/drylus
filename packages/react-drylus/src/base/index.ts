export * from './DrylusProvider';
export * from './ThemeProvider';
export * from './Page';
export * from './Content';
