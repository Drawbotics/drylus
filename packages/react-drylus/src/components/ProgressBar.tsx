import sv from '@drawbotics/drylus-style-vars';
import { css, cx } from 'emotion';
import React from 'react';

import { Color, Size } from '../enums';
import { Responsive, Style } from '../types';
import { getEnumAsClass, useResponsiveProps } from '../utils';
import { placeholderStyles as shimmerStyles } from './LoadingPlaceholder';

const styles = {
  root: css`
    height: ${sv.marginExtraSmall};
    width: 100%;
    position: relative;
    border-radius: ${sv.marginExtraSmall};
    overflow: hidden;
    background: ${sv.neutralLight};

    [data-element='bar'] {
      border-radius: ${sv.marginExtraSmall};
    }
  `,
  bar: css`
    position: absolute;
    z-index: 2;
    top: 0;
    left: 0;
    height: 100%;
    width: 0;
    background: ${sv.neutralDark};
    transition: ${sv.transitionShort};
    overflow: hidden;

    &::after {
      background: ${sv.neutralDark};
      border-radius: 0 !important;
      background-image: linear-gradient(
        to right,
        ${sv.neutralDark} 8%,
        ${sv.neutral} 18%,
        ${sv.neutralDark} 33%
      ) !important;
    }
  `,
  indeterminate: css`
    width: 100% !important;
  `,
  large: css`
    height: ${sv.marginSmall};
    border-radius: ${sv.marginSmall};

    [data-element='bar'] {
      border-radius: ${sv.marginSmall};
    }
  `,
  extraLarge: css`
    height: 90px;
  `,
  small: css`
    height: calc(${sv.marginExtraSmall} / 2);
    border-radius: calc(${sv.marginExtraSmall} / 2);

    [data-element='bar'] {
      border-radius: calc(${sv.marginExtraSmall} / 2);
    }
  `,
  brand: css`
    background: ${sv.brand};

    &::after {
      background: ${sv.brand};
      background-image: linear-gradient(
        to right,
        ${sv.brand} 8%,
        ${sv.brandLight} 18%,
        ${sv.brand} 33%
      ) !important;
    }
  `,
  red: css`
    background: ${sv.red};

    &::after {
      background: ${sv.red};
      background-image: linear-gradient(
        to right,
        ${sv.red} 8%,
        ${sv.redLight} 18%,
        ${sv.red} 33%
      ) !important;
    }
  `,
  blue: css`
    background: ${sv.blue};

    &::after {
      background: ${sv.blue};
      background-image: linear-gradient(
        to right,
        ${sv.blue} 8%,
        ${sv.blueLight} 18%,
        ${sv.blue} 33%
      ) !important;
    }
  `,
  orange: css`
    background: ${sv.orange};

    &::after {
      background: ${sv.orange};
      background-image: linear-gradient(
        to right,
        ${sv.orange} 8%,
        ${sv.orangeLight} 18%,
        ${sv.orange} 33%
      ) !important;
    }
  `,
  green: css`
    background: ${sv.green};

    &::after {
      background: ${sv.green};
      background-image: linear-gradient(
        to right,
        ${sv.green} 8%,
        ${sv.greenLight} 18%,
        ${sv.green} 33%
      ) !important;
    }
  `,
};

export interface ProgressBarProps {
  /** Determines the amount of the bar which is completed, between 0 and 1. If not given the bar is indeterminate */
  percentage?: number;

  /** @kind Color */
  color?: Color.BRAND | Color.RED | Color.BLUE | Color.GREEN | Color.ORANGE;

  /**
   * @default Size.DEFAULT
   * @kind Size
   */
  size?: Size.SMALL | Size.DEFAULT | Size.LARGE;

  /** Used for style overrides */
  style?: Style;

  /** Used for style overrides */
  className?: string;

  /** Reponsive prop overrides */
  responsive?: Responsive<this>;
}

export const ProgressBar = ({ responsive, ...rest }: ProgressBarProps) => {
  const { percentage, size = Size.DEFAULT, style, color, className } = useResponsiveProps<
    ProgressBarProps
  >(rest, responsive);

  const indeterminate = percentage == null;

  return (
    <div
      style={style}
      className={cx(
        styles.root,
        {
          [styles[getEnumAsClass<typeof styles>(size)]]: size != null,
        },
        className,
      )}>
      <div
        data-element="bar"
        className={cx(styles.bar, {
          [styles[getEnumAsClass<typeof styles>(color)]]: color != null,
          [styles.indeterminate]: indeterminate,
          [shimmerStyles.shimmer]: indeterminate,
        })}
        style={{ width: `${(percentage ?? 0) * 100}%` }}
      />
    </div>
  );
};
