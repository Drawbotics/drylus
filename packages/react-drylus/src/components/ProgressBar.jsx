import React from 'react';
import { css, cx, keyframes } from 'emotion';
import PropTypes from 'prop-types';
import sv from '@drawbotics/drylus-style-vars';

import { Categories, Sizes } from '../base';
import { getEnumAsClass } from '../utils';


const translateX = keyframes`
  0% {
    transform: translateX(-40%);
  }
  100% {
    transform: translateX(140%);
  }
`;


const styles = {
  root: css`
    height: ${sv.marginExtraSmall};
    width: 100%;
    position: relative;
    border-radius: ${sv.marginExtraSmall};
    overflow: hidden;
    background: ${sv.neutralLight};

    [data-element="bar"] {
      border-radius: ${sv.marginExtraSmall};
    }
  `,
  bar: css`
    position: absolute;
    z-index: 2;
    top: 0;
    left: 0;
    height: 100%;
    width: 0;
    background: ${sv.neutralDark};
    transition: ${sv.transitionShort};
    overflow: hidden;
  `,
  indeterminate: css`
    width: 50% !important;
    animation: ${translateX}
      calc(${sv.defaultTransitionTime} * 2)
      ease-in-out
      alternate
      infinite;
  `,
  large: css`
    height: ${sv.marginSmall};
    border-radius: ${sv.marginSmall};

    [data-element="bar"] {
      border-radius: ${sv.marginSmall};
    }
  `,
  extraLarge: css`
    height: 90px;
  `,
  small: css`
    height: calc(${sv.marginExtraSmall} / 2);
    border-radius: calc(${sv.marginExtraSmall} / 2);

    [data-element="bar"] {
      border-radius: calc(${sv.marginExtraSmall} / 2);
    }
  `,
  brand: css`
    background: ${sv.brand};
  `,
  danger: css`
    background: ${sv.red};
  `,
  info: css`
    background: ${sv.blue};
  `,
  warning: css`
    background: ${sv.orange};
  `,
  success: css`
    background: ${sv.green};
  `,
};


const ProgressBar = ({
  percentage,
  category,
  size,
  style,
}) => {
  const indeterminate = percentage == null;

  return (
    <div
      style={style}
      className={cx(styles.root, {
        [styles[getEnumAsClass(size)]]: size,
      })}>
      <div
        data-element="bar"
        className={cx(styles.bar, {
          [styles[getEnumAsClass(category)]]: category,
          [styles.indeterminate]: indeterminate,
        })}
        style={{ width: `${percentage * 100}%` }} />
    </div>
  );
};


ProgressBar.propTypes = {
  /** Determines the amount of the bar which is completed, between 0 and 1. If not given the bar is indeterminate */
  percentage: PropTypes.number,

  category: PropTypes.oneOf([
    Categories.BRAND,
    Categories.DANGER,
    Categories.SUCCESS,
    Categories.INFO,
    Categories.WARNING,
  ]),

  size: PropTypes.oneOf([
    Sizes.SMALL,
    Sizes.DEFAULT,
    Sizes.LARGE,
  ]),

  /** Used for style overrides */
  style: PropTypes.object,
};


ProgressBar.defaultProps = {
  size: Sizes.DEFAULT,
};


export default ProgressBar;
