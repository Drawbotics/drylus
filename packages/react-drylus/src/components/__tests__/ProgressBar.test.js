import React from 'react';
import { create } from 'react-test-renderer';

import ProgressBar from '../ProgressBar';
import { Categories, Sizes } from '../../base';


describe('ProgressBar', () => {
  describe('matches snapshot when', () => {
    it('has a value', () => {
      const tree = create(
        <ProgressBar progress={0.4} />
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('has a category', () => {
      const tree = create(
        <ProgressBar progress={0.4} category={Categories.BRAND} />
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('is small', () => {
      const tree = create(
        <ProgressBar progress={0.4} size={Sizes.SMALL} />
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});