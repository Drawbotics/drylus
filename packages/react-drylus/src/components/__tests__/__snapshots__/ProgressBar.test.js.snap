// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar matches snapshot when has a category 1`] = `
@keyframes animation-0 {
  0% {
    -webkit-transform: translateX(-40%);
    -ms-transform: translateX(-40%);
    transform: translateX(-40%);
  }

  100% {
    -webkit-transform: translateX(140%);
    -ms-transform: translateX(140%);
    transform: translateX(140%);
  }
}

.css-1l7orhi-ProgressBar__root {
  height: 8px;
  width: 100%;
  position: relative;
  border-radius: 8px;
  overflow: hidden;
  background: #eaeff4;
}

.css-1l7orhi-ProgressBar__root [data-element="bar"] {
  border-radius: 8px;
}

.css-f49gsh-ProgressBar__bar-ProgressBar__indeterminate {
  position: absolute;
  z-index: 2;
  top: 0;
  left: 0;
  height: 100%;
  width: 0;
  background: #879ba6;
  -webkit-transition: all 0.15s ease-in-out;
  transition: all 0.15s ease-in-out;
  overflow: hidden;
  width: 50% !important;
  -webkit-animation: animation-0 calc(0.3s * 2) ease-in-out alternate infinite;
  animation: animation-0 calc(0.3s * 2) ease-in-out alternate infinite;
}

<div
  className="css-1l7orhi-ProgressBar__root undefined"
>
  <div
    className="undefined css-f49gsh-ProgressBar__bar-ProgressBar__indeterminate"
    data-element="bar"
    style={
      Object {
        "width": "NaN%",
      }
    }
  />
</div>
`;

exports[`ProgressBar matches snapshot when has a value 1`] = `
@keyframes animation-0 {
  0% {
    -webkit-transform: translateX(-40%);
    -ms-transform: translateX(-40%);
    transform: translateX(-40%);
  }

  100% {
    -webkit-transform: translateX(140%);
    -ms-transform: translateX(140%);
    transform: translateX(140%);
  }
}

.css-1l7orhi-ProgressBar__root {
  height: 8px;
  width: 100%;
  position: relative;
  border-radius: 8px;
  overflow: hidden;
  background: #eaeff4;
}

.css-1l7orhi-ProgressBar__root [data-element="bar"] {
  border-radius: 8px;
}

.css-f49gsh-ProgressBar__bar-ProgressBar__indeterminate {
  position: absolute;
  z-index: 2;
  top: 0;
  left: 0;
  height: 100%;
  width: 0;
  background: #879ba6;
  -webkit-transition: all 0.15s ease-in-out;
  transition: all 0.15s ease-in-out;
  overflow: hidden;
  width: 50% !important;
  -webkit-animation: animation-0 calc(0.3s * 2) ease-in-out alternate infinite;
  animation: animation-0 calc(0.3s * 2) ease-in-out alternate infinite;
}

<div
  className="css-1l7orhi-ProgressBar__root undefined"
>
  <div
    className="css-f49gsh-ProgressBar__bar-ProgressBar__indeterminate"
    data-element="bar"
    style={
      Object {
        "width": "NaN%",
      }
    }
  />
</div>
`;

exports[`ProgressBar matches snapshot when is small 1`] = `
@keyframes animation-0 {
  0% {
    -webkit-transform: translateX(-40%);
    -ms-transform: translateX(-40%);
    transform: translateX(-40%);
  }

  100% {
    -webkit-transform: translateX(140%);
    -ms-transform: translateX(140%);
    transform: translateX(140%);
  }
}

.css-1l7orhi-ProgressBar__root {
  height: 8px;
  width: 100%;
  position: relative;
  border-radius: 8px;
  overflow: hidden;
  background: #eaeff4;
}

.css-1l7orhi-ProgressBar__root [data-element="bar"] {
  border-radius: 8px;
}

.css-f49gsh-ProgressBar__bar-ProgressBar__indeterminate {
  position: absolute;
  z-index: 2;
  top: 0;
  left: 0;
  height: 100%;
  width: 0;
  background: #879ba6;
  -webkit-transition: all 0.15s ease-in-out;
  transition: all 0.15s ease-in-out;
  overflow: hidden;
  width: 50% !important;
  -webkit-animation: animation-0 calc(0.3s * 2) ease-in-out alternate infinite;
  animation: animation-0 calc(0.3s * 2) ease-in-out alternate infinite;
}

<div
  className="css-1l7orhi-ProgressBar__root undefined"
>
  <div
    className="css-f49gsh-ProgressBar__bar-ProgressBar__indeterminate"
    data-element="bar"
    style={
      Object {
        "width": "NaN%",
      }
    }
  />
</div>
`;
